src/ape/api/providers.py

```diff
@@ -187,3 +189,95 @@ def snapshot(self) -> str:
     @abstractmethod
     def revert(self, snapshot_id: str):
         ...
+
+
+class Web3Provider(ProviderAPI):
+    """
+    A base provider that is web3 / RPC based.
+    """
+
+    _web3: Web3 = None  # type: ignore
+
+    def connect(self):
+        self._web3 = Web3(HTTPProvider(self.uri))
```

Comment:
what is `self.uri`? perhaps add it as an abstractmethod property

Reply:
Talked elsewhere, plan is to keep `connect` and `disconnect` abstract for this class because not all web3 providers will need a URI